### PR TITLE
Columbus-4 specific rules and log finder

### PR DESCRIPTION
**Pull request by jbamlee**
## Summary

- TerraSwap columbus-4 specific rules

## Checklist

- [ ] Backward compatible?
- [ ] Test enough in your local environment?
- [ ] Add related test cases?

### CODE REVIEW

File: pkg/rules/terraswap/col4_logfinders_test.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+	for idx, tc := range tcs {
+		errMsg := fmt.Sprintf("idx(%d): %s", idx, tc.errMsg)
+		assert := assert.New(t)
+
+		setUp(tc.rawLogStr, tc.pairs, tc.finderFunc)
+		matchedResults := logFinder.FindFromLogs(eventLogs)
+		assert.Len(matchedResults, tc.expectedResultLen, errMsg)
+		if tc.expectedResultLen > 0 {
+			assert.Len(matchedResults[0], tc.matchedLen, "must return all matched value")
```
Comment (psy2848048) [inline pkg/rules/terraswap/col4_logfinders_test.go:94]: The test fails in my local at this line. Is there any fix from the following changes?

File: pkg/rules/terraswap/col4_logfinders.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+			return ok
+		}
+	}
+	rule := col4PairCommonRule
+	rule.Items[PairAddrIdx].Filter = filter
+	return eventlog.NewLogFinder(rule)
+}
+
+// Track cw20 transfer
+func CreateWasmCommonTransferRuleFinder(pairs map[string]bool) (eventlog.LogFinder, error) {
```
Comment (psy2848048) [inline pkg/rules/terraswap/col4_logfinders.go:34]: `CreateWasmCommonTransferRuleFinder` and `CreateTransferRuleFinder` seems to have no test yet

File: pkg/dex/terraswap/client.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	}
+
+	req, err := dex.QueryToJsonStr[dex.FactoryPairsReq](pairsReq)
+	if err != nil {
+		return nil, errors.Wrap(err, "col4QueryClient.QueryPairs")
+	}
+
+	res, err := col4.QueryContractState[dex.FactoryPairsRes](c.lcd, factoryAddr, req, height...)
+	if err != nil {
+		return nil, errors.Wrap(err, "col4QueryClient.QueryPool")
```
[the remark this answers is not in the document]
Comment (jbamlee) [reply pkg/dex/terraswap/client.go]: https://github.com/dezswap/cosmwasm-etl/pull/11/commits/3b0df780d8d424d87bbbebd302900084cf5ff8ec

File: pkg/rules/terraswap/col4_logfinders.go (hunk 2 above)
Comment (jbamlee) [reply pkg/rules/terraswap/col4_logfinders.go:34]: https://github.com/dezswap/cosmwasm-etl/pull/11/commits/f44accf3054f18fc530d41048dc70b3fbc01a00b

File: pkg/rules/terraswap/col4_logfinders_test.go (hunk 1 above)
Comment (jbamlee) [reply pkg/rules/terraswap/col4_logfinders_test.go:98]: Yes, I divided PR for readability.

**Review by jhlee-young — APPROVED** [review]

**Pull request merged**